### PR TITLE
Distribute batch queue jobs per run

### LINKED ISSUE #26: Loop over tasks per dataset
Currently, main.py loops over tasks and then loops over all datasets for each task. 

Instead might be better to loop over all datasets instead, and perform the tasks for each dataset, in case of transfer backlogs.

### PR DESCRIPTION
**Pull request by tunnell**
Adds a command 'massive-cax' that distributes batch queue jobs of cax jobs that run on single runs.

Closes #26.

### CODE REVIEW

File: cax/config.py [hunk 1]
```diff
@@ -160,59 +163,58 @@ def data_availability(hostname=get_hostname()):
...
+    default_args = dict(host=host,
+                        use='cax',
+                        number=333,
+                        ncpus=4,
+                        pax_version=pax.__version__,
+                        partition='xenon1t' if midway else 'main',
+                        base='/project/lgrandi/xenon1t' if midway else '/cfs/klemming/projects/xenon/xenon1t',
+                        account='pi-lgrandi' if midway else 'xenon',
+                        anaconda='/project/lgrandi/anaconda3/bin' if midway else '/afs/pdc.kth.se/projects/xenon/software/Anaconda3r5/bin',
+                        extra='#SBATCH --mem-per-cpu=2000' if midway else '#SBATCH -t 72:00:00',
```
Comment (pdeperio) [inline cax/config.py]: also need ```#SBATCH --qos=xenon1t``` for Midway? or you tested without this and worked?

File: cax/config.py [hunk 2]
```diff
@@ -160,59 +163,58 @@ def data_availability(hostname=get_hostname()):
...
-cd ${{PROCESSING_DIR}}
-rm -f pax_event_class*
+#SBATCH --output={base}/{use}/logs/{number}_{pax_version}_%J.log
+#SBATCH --error={base}/{use}/logs/{number}_{pax_version}_%J.log
+#SBATCH --account={account}
+#SBATCH --partition={partition}
 
-source activate pax_{pax_version}
+#SBATCH --mail-user=42osx7+eo8qpruqa9w8w@sharklasers.com
+#SBATCH --mail-type=ALL
```
Comment (pdeperio) [inline cax/config.py]: should be able to remove all ```--mail*``` configurations and will not send mail

**Comment by pdeperio** [comment]
- Added lcg-cp implementation, including multi-stream and simultaneous LFC registration. 
- Need to decide how to create "midway-login1" DB entries together with the "midway-srm" entries, or have checksum/processing/etc. look for midway-srm when on midway-login1
- When ready, should add "midway-srm" to cax.json upload options for xe1t-datatransfer, and remove download option from midway-login1

File: cax/tasks/data_mover.py [hunk 3]
```diff
@@ -30,18 +31,79 @@ def copy(self, datum_original, datum_destination, method, option_type):
...
+        elif method == 'lcg-cp':
+            self.copyLCGCP(datum_original, datum_destination, server, option_type, nstreams)
 
         else:
             print (method+" not implemented")
             raise NotImplementedError()
 
-    def copyGFAL(self, datum_original, datum_destination, server, option_type):
+    def copyLCGCP(self, datum_original, datum_destination, server, option_type, nstreams):
+        """Copy data via GFAL function
```
Comment (tunnell) [inline cax/tasks/data_mover.py]: gfal

**Comment by tunnell** [comment]
Here's what I'm thinking.  Name would be midway.  Method would be a list.  Then copy would prioritize the grid transfer, if available.

File: cax/tasks/data_mover.py [hunk 4]
```diff
@@ -30,18 +31,79 @@ def copy(self, datum_original, datum_destination, method, option_type):
             server = config_original['hostname']
             username = config_original['username']
 
+        nstreams = 16
+
         # Determine method for remote site
         if method == 'scp':
```
Comment (tunnell) [inline cax/tasks/data_mover.py]: Here, we'd check `if 'scp' in methods`.
We should check GRID first, otherwise scp.

**Comment by tunnell** [comment]
Here is the proposal.  This can be a new pull request or here.  

- Midway gets renamed just 'midway' for both SRM and login.
- The GRID location becomes an extra field 'url' that data_mover LGC fetches.
- We make 'method' a list.  In data mover, the check becomes `if 'lgc' in methods` then do GRID, then same for SCP.  The order of this will be the order of priority.  
- When we run cax at Midway, we do `HOSTNAME=midway cax`

**Pull request merged**